tests/techmap/shiftx2mux.ys

```diff
@@ -104,6 +104,7 @@ wire [7:0] AA = {1'bx, A};
 assign Y = AA[B*2 +: 2];
 endmodule
 EOT
+proc
```

Comment:
Why the `proc` here? I'm asking because that testcase isn't directly related to the new attribute being added and also doesn't have any processes.

Reply:
IIRC I had to add proc to get past some error back when I worked on this, presumably because I moved transformations from RTLIL to AST.

It seems proc is no longer needed now , so I've removed it.

Reply:
Bah, that still failed on two architectures; test-macos (macos-11, Big Sur, c++11) and test-linux (ubuntu-20.04, focal, gcc-11, c++11):

Warning: Ignoring module top because it contains processes (run 'proc' command first).
ERROR: Gold module contains memories or processes. Run 'memory' or 'proc' respectively.

Reply:
Huh, can the AST transformation add processes where there weren't any originally?

Reply:
(FWIW I think those two architectures are the only ones where tests are run, for the other platforms the CI only does a test build, even though it's all confusingly named 'Build and run tests').

Reply:
> Huh, can the AST transformation add processes where there weren't any originally?

An AST_CASE as generated by `(* nordshift *)` must be within an AST_ALWAYS process, however the default AST_SHIFTX can be used directly in an `assign`.

I have now reworked the code so that a temporary identifier (and thus a process) is only generated for `(* nordshift *)`.
This aligns the default behavior with the previous behavior of `genrtlil.c`.

It can be instructive to run `yosys -p "read_verilog -dump_vlog2 top.v"` on the module which failed previously, both with and without `(* nordshift *)`:

```
module top(input [6:0] A, input [1:0] B, output [1:0] Y);
//(* nordshift *)
wire [7:0] AA = {1'bx, A};
assign Y = AA[B*2 +: 2];
endmodule
```